### PR TITLE
Fixed Bug that would clear a Gallery

### PR DESCRIPTION
**Pull request by janpetersen**
I am targeting this branch, because I use it.

## Changelog

```markdown
### Fixed
- Fixed Bug that would clear a Galery
- Selected gallery context remained untranslated in dropdown
```

## Subject

When saving an existing gallery, all assigned images were removed:

DELETE FROM media__gallery_media WHERE gallery_id = 1;

This was due to replacing the PersistentCollection with a fresh new ArrayCollection, probably without proper cascade=persist on Doctrine level.

### CODE REVIEW

File: src/Model/Gallery.php [hunk 1]
```diff
@@ -144,8 +144,6 @@ public function getDefaultFormat()
      */
     public function setGalleryHasMedias($galleryHasMedias)
     {
-        $this->galleryHasMedias = new ArrayCollection();
-
```
Comment (greg0ire) [inline src/Model/Gallery.php]: Shouldn't the collection be cleared first? There might be things to remove in there.

Comment (janpetersen) [reply src/Model/Gallery.php]: I don't know why we should clear the collection first.

Deleting an image from the Gallery works just fine.

Comment (greg0ire) [reply src/Model/Gallery.php]: Because anyone might call `setGalleryHasMedias` with any custom code expect a replacement to happen, not a merge.

File: src/Resources/views/GalleryAdmin/list.html.twig [hunk 2]
```diff
@@ -18,7 +18,7 @@ file that was distributed with this source code.
                 {{ "label.select_context"|trans({}, 'SonataMediaBundle') }}
                 <div class="btn-group">
                     <button type="button" class="btn btn-default dropdown-toggle" data-toggle="dropdown">
-                        <strong class="text-info">{{ persistent_parameters.context }}</strong> <span class="caret"></span>
+                        <strong class="text-info">{{ persistent_parameters.context|trans({}, 'SonataMediaBundle') }}</strong> <span class="caret"></span>
```
Comment (jordisala1991) [inline src/Resources/views/GalleryAdmin/list.html.twig]: How is this related to the fix?

File: src/Model/Gallery.php (hunk 1 above)
Comment (janpetersen) [reply src/Model/Gallery.php]: Why would anyone want to replace all images of an existing Gallery with a complete set of new images? In that case, he should better add a custom method like "replaceGalleryHasMedia", since he most probably will have the Gallery model extended anyways.

This issue just removed all images from my local development instance of a website. I think you should focus on fixing this before someone experiences some real trouble.

File: src/Resources/views/GalleryAdmin/list.html.twig (hunk 2 above)
Comment (janpetersen) [reply src/Resources/views/GalleryAdmin/list.html.twig]: Sorry, I did not see that. It was part of a pull request from December that I thought were already merged... https://github.com/sonata-project/SonataMediaBundle/pull/1345 ?

File: src/Model/Gallery.php (hunk 1 above)
Comment (greg0ire) [reply src/Model/Gallery.php]: > Why would anyone want to replace all images of an existing Gallery with a complete set of new images? In that case, he should better add a custom method like "replaceGalleryHasMedia", since he most probably will have the Gallery model extended anyways.

I don't know, you don't know, let's not take any chances, ok? I don't know why you insist on removing this, you or @gremo? The `Collection` interface has a `clear` method: https://github.com/doctrine/collections/blob/master/lib/Doctrine/Common/Collections/Collection.php#L43

Just call, let me approve this PR, and get that bug fixed.

File: src/Resources/views/GalleryAdmin/list.html.twig (hunk 2 above)
Comment (greg0ire) [reply src/Resources/views/GalleryAdmin/list.html.twig]: Do you know how to get rid of this with `git rebase -i`?

File: src/Model/Gallery.php (hunk 1 above)
Comment (greg0ire) [reply src/Model/Gallery.php]: Oh sorry I get it now… not sure how to solve this.

**Comment by gremo** [comment]
Why we don't talk about this PR in the original https://github.com/sonata-project/SonataMediaBundle/pull/1370
